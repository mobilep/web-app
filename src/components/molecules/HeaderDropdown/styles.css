.HeaderDropdown {
	--boxShadowColor1: rgba(0, 0, 0, .1);
	--boxShadowColor2: rgba(0, 0, 0, .3);

	position: relative;
}

.HeaderDropdown-dotsButton {
	color: var(--tomatoRed);
	overflow: hidden;
}

.HeaderDropdown-dropDownWrapper {
	position: absolute;
	background: var(--white);
	box-shadow: 0 1px 5px var(--boxShadowColor2);
	width: 190px;
	top: 45px;
	left: -160px;
	z-index: 1;
}

.HeaderDropdown .HeaderDropdown-dropDownWrapper {
	display: none;
}

.HeaderDropdown-dropDownVisible .HeaderDropdown-dropDownWrapper {
	display: block;
}

.HeaderDropdown-dropDown {
	position: relative;
	display: flex;
	flex-direction: column;
}

.HeaderDropdown-triangle {
	width: 26px;
	height: 15px;
	position: absolute;
	background: var(--white);
	right: 5px;
	top: -12px;
	overflow: hidden;
}

.HeaderDropdown-triangle::after {
	content: '';
	width: 26px;
	height: 26px;
	position: absolute;
	left: 0;
	top: 5px;
	transform: rotate(45deg);
	box-shadow: -1px -1px var(--boxShadowColor1);
	border-radius: 3px;
}

.HeaderDropdown-button {
	font-size: 16px;
	min-height: 45px;
	border-radius: 0;
}

.HeaderDropdown-button + .HeaderDropdown-button {
	margin-top: 3px;
	border-top: 1px solid var(--tomatoRed);
}

.HeaderDropdown-attach {
	width: 50px;
	height: 50px;
	border: none;
	background-color: transparent;
	padding: 0;
	color: var(--grey3);
}

.HeaderDropdown-attach svg {
	width: 50px;
	height: 50px;
	object-fit: contain;
	position: absolute;
	left: 14px;
	top: 14px;
}

.HeaderDropdown-attach:hover {
	color: var(--grey3);
}

.HeaderDropdown-attach:focus {
	color: var(--grey3);
}

.HeaderDropdown-wrapper-upside-down {
	left: -140px;
	bottom: calc(100% + 15px);
	top: unset;
}

.HeaderDropdown-triangle-down {
	top: calc(100% - 2px);
	right: 10px;
	transform: rotate(180deg);
}
