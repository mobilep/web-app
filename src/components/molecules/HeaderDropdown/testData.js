export default [
	{
		label: 'edit',
		onClick: () => {},
	},
	{
		label: 'delete',
		onClick: () => {},
	},
];
