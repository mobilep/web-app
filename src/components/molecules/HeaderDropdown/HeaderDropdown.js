import React, { Component, createRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Button, DotsButton } from '../../atoms';
import './styles.css';
import { AttachIcon } from '../../../icons';

export default class HeaderDropdown extends Component {

	static propTypes = {
		attachMenu: PropTypes.bool,
		buttonsProps: PropTypes.arrayOf(PropTypes.shape({
			label: PropTypes.string.isRequired,
			onClick: PropTypes.func.isRequired,
		})).isRequired,
	}

	state = {
		show: false,
	}

	ref = createRef();

	componentDidMount () {
		document.addEventListener('click', this.handleDocumentClick);
	}

	componentWillUnmount () {
		document.removeEventListener('click', this.handleDocumentClick);
	}

	handleDocumentClick = (event) => {
		const { target } = event;
		const { show } = this.state;
		if (!this.ref.current) return;
		const isDropDownAreaClick = this.ref.current.contains(target);
		if (!show || isDropDownAreaClick) return;
		this.setState({ show: false });
	}

	handleClick = () => {
		const { show } = this.state;
		this.setState({ show: !show });
	}

	getClassName ({ show } = this.state) {
		return classNames({
			'HeaderDropdown-dropDownVisible': show,
		}, 'HeaderDropdown');
	}

	getDropDownButtonsFromProps ({ buttonsProps } = this.props) {
		return buttonsProps
			.map(({ label, onClick }) =>
				(<Button
					key={label}
					className='HeaderDropdown-button'
					color='white'
					onClick={onClick}
				>
					{label}
				</Button>));
	}

	getDropdownWrappperClassName (attachMenu) {
		return classNames({
			'HeaderDropdown-dropDownWrapper': true,
			'HeaderDropdown-wrapper-upside-down': attachMenu,
		});
	}

	getTriangleClassName (attachMenu) {
		return classNames({
			'HeaderDropdown-triangle': true,
			'HeaderDropdown-triangle-down': attachMenu,
		});
	}

	render () {
		const className = this.getClassName();
		return (
			<div ref={this.ref} className={className}>
				{this.props.attachMenu
					? <Button className='HeaderDropdown-attach' onClick={this.handleClick}><AttachIcon /></Button>
					: <DotsButton onClick={this.handleClick} className='HeaderDropdown-dotsButton' />}
				<div className={this.getDropdownWrappperClassName(this.props.attachMenu)}>
					<div className='HeaderDropdown-dropDown'>
						{this.getDropDownButtonsFromProps()}
					</div>
					<div className={this.getTriangleClassName(this.props.attachMenu)} />
				</div>
			</div>
		);
	}

}
