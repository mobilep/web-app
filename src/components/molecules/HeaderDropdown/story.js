import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';

import HeaderDropdown from './HeaderDropdown';
import buttonsProps from './testData';

storiesOf('2. Molecules / HeaderDropdown', module)
	.add('HeaderDropdown', () => (
		<div style={{ marginLeft: '190px', marginTop: '150px' }}>
			<HeaderDropdown demo buttonsProps={buttonsProps} attachMenu={boolean('Attach menu', false)} />
		</div>
	));
