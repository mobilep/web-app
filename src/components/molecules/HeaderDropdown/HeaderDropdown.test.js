import React from 'react';
import { shallow } from 'enzyme';

import HeaderDropdown from './HeaderDropdown';

import buttonsProps from './testData';

describe('Molecule - HeaderDropdown', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<HeaderDropdown buttonsProps={ buttonsProps} />
		);
		expect(wrapper).toMatchSnapshot();
	});

});
