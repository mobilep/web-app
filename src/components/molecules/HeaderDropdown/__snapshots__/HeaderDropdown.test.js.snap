// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - HeaderDropdown Should render 1`] = `
<div
  className="HeaderDropdown"
>
  <DotsButton
    className="HeaderDropdown-dotsButton"
    onClick={[Function]}
  />
  <div
    className="HeaderDropdown-dropDownWrapper"
  >
    <div
      className="HeaderDropdown-dropDown"
    >
      <Button
        className="HeaderDropdown-button"
        color="white"
        key="edit"
        onClick={[Function]}
      >
        edit
      </Button>
      <Button
        className="HeaderDropdown-button"
        color="white"
        key="delete"
        onClick={[Function]}
      >
        delete
      </Button>
    </div>
    <div
      className="HeaderDropdown-triangle"
    />
  </div>
</div>
`;
